File: src/cotestudy/완전탐색/no16283.java
package cotestudy.완전탐색;
import java.io.*;
import java.util.*;
public class no16283 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        int n = Integer.parseInt(st.nextToken());
        int w = Integer.parseInt(st.nextToken());
        int sheep = 0;
        int goat = 0;
        int cnt = 0;

        for (int i = 1; i < n; i++) {
            if ((a - b) * i + b * n == w) {
                sheep = i;
                goat = n - sheep;
                cnt++;
            }
        }
        if(cnt == 1){
            bw.write(sheep + " " + goat);
        }
        else
            bw.write("-1");
        bw.flush();
        bw.close();
    }
}

File: src/cotestudy/완전탐색/no16283.py
a, b, n, w = map(int, input().split())
cnt = 0
sheep = 0
goat = 0
for x in range(1, n):
    if a*x + b*(n-x) == w:
        cnt += 1
        sheep = x
        goat = n-x

if cnt== 1 :
    print(sheep,goat)

else:
    print(-1)


